lib/statsd/instrument/assertions.rb

```diff
@@ -31,41 +31,56 @@ def assert_statsd_key_value(metric_name, options = {}, &block)
-    assert options[:times] === metrics.length, "The amount of StatsD calls for metric #{metric_name} was unexpected. Expected #{options[:times].inspect}, found #{metrics.length}"
 
-    metrics.each do |metric|
+    expected_metrics.each do |expected_metric|
+      matching_metrics = metrics.select { |m| m.type == expected_metric.type && m.name == expected_metric.name }
+      assert matching_metrics.length > 0, "No StatsD calls for metric #{expected_metric.name} were made."
 
-      assert within_numeric_range?(metric.sample_rate), "Unexpected sample rate type for metric #{metric_name}, must be numeric"
-      assert_equal options[:sample_rate], metric.sample_rate, "Unexpected StatsD sample rate for metric #{metric_name}" if options[:sample_rate]
-      assert_equal options[:value], metric.value, "Unexpected value submitted for StatsD metric #{metric_name}" if options[:value]
+      matching_metrics.each do |actual_metric|
+        assert_metric_matches_expectation(actual_metric, expected_metric)
+      end
 
-      if options[:tags]
+      assert expected_metric.times == matching_metrics.length,
+        "The amount of StatsD calls for metric #{expected_metric.name} was unexpected. Expected #{expected_metric.times}, found #{matching_metrics.length}"
+    end
+  end
 
-        expected_tags = Set.new(StatsD::Instrument::Metric.normalize_tags(options[:tags]))
-        actual_tags = Set.new(metric.tags)
+  private
 
-        if options[:ignore_tags]
-          ignored_tags = Set.new(StatsD::Instrument::Metric.normalize_tags(options[:ignore_tags])) - expected_tags
-          actual_tags -= ignored_tags
+  def assert_metric_matches_expectation(actual_metric, expected_metric)
+    assert within_numeric_range?(actual_metric.sample_rate), "Unexpected sample rate type for metric #{actual_metric.name}, must be numeric"
+    assert_equal expected_metric.sample_rate, actual_metric.sample_rate, "Unexpected StatsD sample rate for metric #{actual_metric.name}" if expected_metric.sample_rate
+    assert_equal expected_metric.value, actual_metric.value, "Unexpected value submitted for StatsD metric #{actual_metric.name}" if expected_metric.value
 
-          if options[:ignore_tags].is_a?(Array)
-            actual_tags.delete_if{ |key| options[:ignore_tags].include?(key.split(":").first) }
-          end
-        end
+    if expected_metric.tags
 
-        assert_equal expected_tags, actual_tags,
-                     "Unexpected StatsD tags for metric #{metric_name}. Expected: #{expected_tags.inspect}, actual: #{actual_tags.inspect}"
+      expected_tags = Set.new(StatsD::Instrument::Metric.normalize_tags(expected_metric.tags))
+      actual_tags = Set.new(actual_metric.tags)
+
+      if expected_metric.ignore_tags
+        ignored_tags = Set.new(StatsD::Instrument::Metric.normalize_tags(expected_metric.ignore_tags)) - expected_tags
+        actual_tags -= ignored_tags
+
+        if expected_metric.ignore_tags.is_a?(Array)
```

Comment:
actually this is your change. In the original code, this was comparing to the unnormalized list of tags and now you are comparing it to the normalized version. Is that intended? Does that keep the original behaviour?